packages/protocol/contracts/governance/Election.sol

```diff
@@ -1043,6 +1121,25 @@ contract Election is
     return res;
   }
 
+  /**
+   * @notice Allows to turn on/off voting over maxNumGroupsVotedFor.
+   * Once this is turned on and account voted for more than maxNumGroupsVotedFor,
+   * it is account's obligation to run updateTotalVotesByAccountForGroup, updateTotalVotesByAccount
+   * once a day.
+   * @param flag The on/off flag.
+   */
+  function setAllowedToVoteOverMaxNumberOfGroups(bool flag) public {
+    if (!flag) {
+      require(
+        votes.groupsVotedFor[msg.sender].length < maxNumGroupsVotedFor,
```

Comment:
Rather than just using `msg.sender`, should we allow the use of the vote signer? I think that would boil down to making this mapping `account => cache` rather than `msg.sender => cache`, and looking up the account with `Accounts.voteSignerToAccount` (but double check this is correct). In particular, I think in other parts of these changes we look up `cache[account]` rather than `msg.sender`, so users might get different results when using their main account address or vote signer.